src/ray/raylet/local_task_manager.cc

```diff
@@ -778,6 +781,31 @@ bool LocalTaskManager::CancelTask(
   return false;
 }
 
+void LocalTaskManager::CancelInfeasiblePlacementGroupTasks(
+    std::deque<std::shared_ptr<internal::Work>> &tasks) {
+  for (auto work_it = tasks.begin(); work_it != tasks.end();) {
+    if ((*work_it)->GetState() != internal::WorkStatus::WAITING) {
+      // Only cancel waiting tasks.
+      ++work_it;
+      continue;
+    }
+    if ((*work_it)->task.GetTaskSpecification().PlacementGroupBundleId().first ==
```

Comment:
I think we should move this up, right before calling `CancelInfeasiblePlacementGroupTasks()`. We group tasks by scheduling class, so it's all placement group tasks or none placement group tasks?

Also I think it should be a CHECK to reflect that it's the only cause now:

I think it should be something like
```
if (infeasible) {
CHECK(worker->IsplacementGroupTask());
CancelInfeasiblePlacementGroupTasks();
}
```